src/server/game/Spells/Spell.cpp

```diff
@@ -5831,46 +5831,69 @@ SpellCastResult Spell::CheckRange(bool strict)
     if (!strict && m_casttime == 0)
```

Comment:
I rather not extend the scope of this PR. Feel free to open a new PR if you would like. Then things can be discussed and tested more thoroughly there.